src/lib/theme/theme.service.ts

```diff
@@ -62,6 +61,18 @@ export class VantageThemeService {
 
     // apply theme on storage or media query change
     merge(storageObserver, mediaObserver).subscribe((theme: VantageTheme) => this.applyTheme(theme));
+
+    // account for cached navigation
+    // needed for Firefox BFCache
+    window.onpageshow = (event: PageTransitionEvent) => {
```

Comment:
Let's use addEventListener instead of this assignment. Also there is no point in checking is persisted and it differs right? I would keep it simple and just always call applyTheme when this event hits. Also we can wrap the accessing of the localstorage and application of theme in a small function that we can re-use here and in the constructor.